tests/beaker_tests/cisco_interface/test_vlan_mapping.rb

```diff
@@ -0,0 +1,186 @@
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# -----------------------------
+# Common settings and variables
+# -----------------------------
+testheader = 'Resource cisco_interface: vlan_mapping properties'
+
+# The 'tests' hash is used to define all of the test data values and expected
+# results. It is also used to pass optional flags to the test methods when
+# necessary.
+
+# 'tests' hash
+# Top-level keys set by caller:
+# tests[:master] - the master object
+# tests[:agent] - the agent object
+# tests[:config_bridge_domain] - the bridge-domain configuration
+# tests[:config_switchport] - the interface switchport configuration
+#
+tests = {
+  master:               master,
+  agent:                agent,
+  testheader:           testheader,
+  config_bridge_domain: 'system bridge-domain 100-113 ; bridge-domain 100',
+  config_switchport:    'switchport ; switchport mode trunk',
+}
+
+# tests[id] keys set by caller and used by test_harness_common:
+#
+# tests[id] keys set by caller:
+# tests[id][:desc] - a string to use with logs & debugs
+# tests[id][:manifest] - the complete manifest, as used by test_harness_common
+# tests[id][:resource] - a hash of expected states, used by test_resource
+# tests[id][:resource_cmd] - 'puppet resource' command to use with test_resource
+# tests[id][:ensure] - (Optional) set to :present or :absent before calling
+# tests[id][:code] - (Optional) override the default exit code in some tests.
+#
+
+tests['default_properties'] = {
+  desc:           '1.1 Default Properties',
+  manifest_props: {
+    vlan_mapping_enable: 'default',
+    vlan_mapping:        'default',
+  },
+  resource:       {
+    'vlan_mapping_enable' => 'true',
+    # 'vlan_mapping' is nil when default
+  },
+}
+
+vlan_maps = [%w(20 21), %w(30 31)]
+tests['non_default_properties'] = {
+  desc:           '2.1 Non Default Properties',
+  manifest_props: {
+    vlan_mapping_enable: 'false',
+    vlan_mapping:        vlan_maps,
+  },
+  resource:       {
+    'vlan_mapping_enable' => 'false',
+    'vlan_mapping'        => "#{vlan_maps}",
+  },
+}
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+def build_manifest_vlan_mapping(tests, id)
+  intf = tests[:intf]
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node default {
+    cisco_interface { '#{intf}':\n
+    #{prop_hash_to_manifest(tests[id][:manifest_props])}
+  }\n      }\nEOF"
+
+  cmd = PUPPET_BINPATH + "resource cisco_interface '#{intf}'"
+  tests[id][:resource_cmd] =
+    get_namespace_cmd(agent, cmd, options)
+end
+
+def test_harness_vlan_mapping(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+
+  # Build the manifest for this test
+  build_manifest_vlan_mapping(tests, id)
+
+  # Workaround for (ioctl) facter bug on n7k ***
+  tests[id][:code] = [0, 2]
+
+  test_harness_common(tests, id)
```

Comment:
We need to remove the `test_show_cmd` from `test_harness_common`

https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/lib/utilitylib.rb#L183

Go ahead and take care of that now?